cmake/CompilerOptions.cmake

```diff
@@ -4,6 +4,8 @@ set(CXXOPT_DEBUG "-g")
 set(CXXOPT_PTHREAD "-pthread")
 set(CXXOPT_CXX11 "-std=c++11")
 set(CXXOPT_WALL "-Wall")
+set(BSCOPT_EXTRAE "-O -g -I $(EXTRAE_HOME)/include")
```

Comment:
These options have three bits of information
    1. debug flags `-g`
    2. optimization flags `-O`
    3. an include path

When `WITH_EXTRAE` is set in the main CMakeLists, you should check if `-g` has been added to `CMAKE_CXX_FLAGS`, and add it if needed.

You should not specify the optimization flag, which might conflict with the default set by the system.

Set the include path using the CMake ` include_directories(${EXTRAE_INCLUDE})` in the main CMakeLists.txt. Make the user provide `EXTRAE_INCLUDE` (maybe an `EXTRAE_ROOT` like for TBB)